File: HackerRank 30 Days/day17/main.cs
using System;

class Calculator
{
    public int power(int n, int p)
    {
        if (n < 0 || p < 0)
        {
            throw new ArgumentException("n and p should be non-negative");
        }

        return (int)Math.Pow(n, p);
    }
}

class Solution
{
    static void Main(string[] args)
    {
        int t = int.Parse(Console.ReadLine() ?? "0");
        Calculator calculator = new Calculator();

        for (int i = 0; i < t; i++)
        {
            string[] values = (Console.ReadLine() ?? "").Split(' ');
            int n = int.Parse(values[0]);
            int p = int.Parse(values[1]);

            try
            {
                Console.WriteLine(calculator.power(n, p));
            }
            catch (ArgumentException ex)
            {
                Console.WriteLine(ex.Message);
            }
        }
    }
}

File: HackerRank 30 Days/day17/main.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', () => {
    inputString = inputString.trim().split('\n');
    main();
});

function readLine() {
    return inputString[currentLine++];
}

class Calculator {
    power(n, p) {
        if (n < 0 || p < 0) {
            throw new Error("n and p should be non-negative");
        }

        return n ** p;
    }
}

function main() {
    const myCalculator = new Calculator();
    const T = parseInt(readLine());

    for (let i = 0; i < T; i++) {
        const [n, p] = readLine().split(' ').map(Number);

        try {
            console.log(myCalculator.power(n, p));
        } catch (e) {
            console.log(e.message);
        }
    }
}